ledger/narwhal/transmission-id/src/lib.rs

```diff
@@ -27,41 +27,50 @@ pub enum TransmissionID<N: Network> {
     /// A ratification.
     Ratification,
     /// A solution.
-    Solution(SolutionID<N>),
+    Solution(SolutionID<N>, N::TransmissionChecksum),
     /// A transaction.
-    Transaction(N::TransactionID),
+    Transaction(N::TransactionID, N::TransmissionChecksum),
 }
 
-impl<N: Network> From<SolutionID<N>> for TransmissionID<N> {
+impl<N: Network> From<(SolutionID<N>, N::TransmissionChecksum)> for TransmissionID<N> {
     /// Converts the solution ID into a transmission ID.
-    fn from(solution_id: SolutionID<N>) -> Self {
-        Self::Solution(solution_id)
+    fn from((solution_id, checksum): (SolutionID<N>, N::TransmissionChecksum)) -> Self {
+        Self::Solution(solution_id, checksum)
     }
 }
 
-impl<N: Network> From<&N::TransactionID> for TransmissionID<N> {
+impl<N: Network> From<(&N::TransactionID, &N::TransmissionChecksum)> for TransmissionID<N> {
```

Comment:
Nit: this impl uses references, while the From impl for solution_id uses values

Reply:
I was unable to do this because the rust type system does not allow multiple impl of the same trait with the same type ownership:

```
error[E0119]: conflicting implementations of trait `From<(snarkvm_ledger_puzzle::SolutionID<_>, &_)>` for type `TransmissionID<_>`
  --> ledger/narwhal/transmission-id/src/lib.rs:42:1
   |
35 | impl<N: Network> From<(SolutionID<N>, N::TransmissionChecksum)> for TransmissionID<N> {
   | ------------------------------------------------------------------------------------- first implementation here
...
42 | impl<N: Network> From<(N::TransactionID, N::TransmissionChecksum)> for TransmissionID<N> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `TransmissionID<_>
   ```